frontend/src/components/pages/ManageUsersPage.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from "react";
+import { useQuery } from "@apollo/client";
+import { Box, Button, Flex, HStack, Text } from "@chakra-ui/react";
+import { DownloadIcon, SearchIcon } from "@chakra-ui/icons";
+import {
+  AdminPage,
+  UserCardProps,
+  UsersByEmail,
+} from "../../types/AdminDashboardTypes";
+import UserCard from "../admin/UserCard";
+import SideBar from "../admin/SideBar";
+import SearchUsersBar from "../admin/SearchUsersBar";
+import { UserResponse } from "../../APIClients/types/UserClientTypes";
+import { USERS } from "../../APIClients/queries/UserQueries";
+
+const NoUserSelectedCard = (): React.ReactElement => {
+  return (
+    <Box
+      align="center"
+      py={20}
+      border="1px"
+      borderColor="background.lightgrey"
+      borderRadius="6px"
+    >
+      <SearchIcon h="100px" w="100px" color="text.grey" />
+      <Text variant="display-md">No User Selected</Text>
+      <Text variant="body">
+        Use the search bar above to view specific user details
+      </Text>
+    </Box>
+  );
+};
+
+const ManageUsersPage = (): React.ReactElement => {
+  const [
+    spotlightedUser,
+    setSpotlightedUser,
+  ] = useState<UserCardProps | null>();
+
+  // Never expose this
+  const [users, setUsers] = useState<UsersByEmail>({});
+
+  useQuery<{
+    users: Array<UserResponse>;
+  }>(USERS, {
```

Comment:
So we are retrieving all the users on load, and then searching through the response when the user types? Does this mean all of the search logic is happening on the frontend?

Not the biggest fan of this, because even though this is a protected route:
1. The admin will have access to all user emails if they try (not a big at all, but idealogically maybe bad)
2. doesn't this mean we have to read all the mongo db users whenever this page loads? Seems impractical

Anyway, this is all to say, why aren't we using the `getUserByEmail` endpoint? This API seems perfect for this application, unless I'm misunderstanding something. That way, it's more efficient, and all of the backend/search functionality can be in the `SearchUsersBar` component

Reply:
Yes that's exactly what's happening right now. It doesn't feel like good practice -- and probably isn't -- but at least for the points you mentioned I still think it makes more sense:
- the admin will have access to all user emails no matter if the search in the frontend or the backend the choice of where we're storing everything does not change access at all
  - to me, that's something that's reasonably within the admin's capabilities
- yes, we will have to read all mongo db users whenever the page loads and I agree it's not the most practical, but isn't using the `getUserByEmail` query even worse? You'd be submitting an API call on every keystroke 
  - even if we sub this with something debounced on 50ms for eg., you'd be submitting a request every 50ms still, which is more frequent than page refreshes
  - pretty sure the `getUserByEmail` query would read through all the documents too? not saving us anything there
 
Is there another solution that I'm missing here?

Reply:
OK discussed at work session, satisfied to move to `getUserByEmail`